src/components/Extensions/Details/DetailCard/Tabs/EventsConfig.tsx

```diff
@@ -25,100 +27,102 @@ import { ParamValue } from './ParamValue';
 
 export function EventsConfig() {
   const extension = useExtension()!;
-  const eventsByPrefix: Map<string, string[]> = new Map(); // maps prefix ==> list of types
-  const eventDescriptionMapping: Map<string, string> = new Map(); // maps type ==> description
-  let allowedEventTypesDiv;
-  let channelLocationDiv;
-  !!extension.events &&
-    extension.events.forEach((event) => {
-      eventDescriptionMapping.set(event.type, event.description);
-    });
-  if (!!extension.allowedEventTypes && extension.allowedEventTypes.length > 0) {
-    extension.allowedEventTypes.forEach((eventType) => {
-      const prefix = eventType.split('.').slice(0, -1).join('.');
-      if (!eventsByPrefix.get(prefix)) {
-        eventsByPrefix.set(prefix, [eventType]);
-      } else {
-        eventsByPrefix.get(prefix)?.push(eventType);
-      }
-    });
-    const allowedEventTypeElements = Array.from(eventsByPrefix.keys())
-      .sort()
-      .map((prefix) => {
-        const events = eventsByPrefix.get(prefix)?.map((event) => {
-          return (
-            <div key={`${event}`} className={styles.eventValue}>
-              <ParamValue value={event} type={ParamType.STRING} />
-              <Typography use="body1" theme="secondary">
-                {eventDescriptionMapping.get(event) || 'Not Specified'}
-              </Typography>
-            </div>
-          );
-        });
+  const eventsByType = !!extension.events
+    ? keyBy(extension.events, 'type')
+    : {};
```

Comment:
```suggestion
  const eventsByType = keyBy(extension.events ?? [], 'type');
```

Also, is this type safe?

Reply:
I'm not super familiar with lodash; the second argument passed in is the string-representation of the property that you're plucking, if you pass a random string, does that break the compiler? If not, I'm not sure we should use it

Reply:
Gotcha. Just made it type safe, good catch.